test/functional/sqlsrv/MsHelper.inc

```diff
@@ -410,6 +539,201 @@ function fetchAll($conn, $tbname)
+}
+
+/**
+ * Insert the specified number of rows with test data into the test table
+ * @param  resource  $conn : connection resource
+ * @param  string  $tbname : name of the table to create
+ * @param  int  $rowCount : number of test rows to be inserted
+ * @return number of rows inserted
+ */
+function insertTestRows($conn, $tbame, $rowCount)
+{
+    $count = 0;
+    for ($i = 0; $i < $rowCount; $i++) {
+        if (insertTestRow($conn, $tbame, rand(1, 20))) {
+            $count++;
+        }
+    }
+
+    if ($count != $rowCount) {
+        die("$count rows inserted instead of $rowCount\n");
+    }
+    return ($count);
+}
+
+/**
+ * Insert a number of rows with test data into the test table given the indexes
+ * @param  resource  $conn : connection resource
+ * @param  string  $tbname : name of the table to create
+ * @param  int  $minIndex : index of the first row to be inserted
+ * @param  int  $maxIndex : index of the last row to be inserted
+ * @return number of rows inserted
+ */
+function insertTestRowsByRange($conn, $tbame, $minIndex, $maxIndex)
+{
+    $count = 0;
+    $rowCount = $maxIndex - $minIndex + 1;
+    if ($rowCount > 0) {
+        for ($i = $minIndex; $i <= $maxIndex; $i++) {
+            if (insertTestRow($conn, $tbame, $i)) {
+                $count++;
+            }
+        }
+    }
+    if ($count != $rowCount) {
+        die("$count rows inserted instead of $rowCount\n");
+    }
+    return ($count);
+}
+
+/**
+ * Insert one or more specific rows of test data into the test table
+ * @param  resource  $conn : connection resource
+ * @param  string  $tbname : name of the table to create
+ * @param  int  $index : the index of a certain row of test data
+ * @return object sqlsrv statement upon success or false otherwise
+ */
+function insertTestRow($conn, $tbname, $index)
+{
+    if ($index < 1 || $index > 20) {
+        echo("Invalid row index $index for test data!\n");
+        return false;
+    }
+
+    // get array of input values
+    $inputArray = getInsertArray($index);
+    if (empty($inputArray)) {
+        fatalError("getInsertSqlComplete: inputs for inserting a row cannot be empty");
+    }
+    
+    $result = null;
+    if (isColEncrypted()) {
+        $stmt = insertRow($conn, $tbname, $inputArray, $result, INSERT_PREPARE_PARAMS);
+    } else {
+        // do not use the generic insertRow as the binary data needs some pre-processing
+        // before calling sqlsrv_query()
+        $colStr = "INSERT INTO $tbname (";
+        $valStr = "VALUES (";
+        $col = 1;
+        foreach ($inputArray as $key => $value) {
+            $colStr .= $key . ", ";
+            
+            if (is_array($value)) {
+                $value = $value[0];
+                // this might be an input to a decimal, a numeric or a binary field
+                if (isBinary($col)) {
+                    $value = "0x" . $value;    // annotate the input string as a hex string
+                }
```

Comment:
can't you call insertRow after you've modified the binary data?

Reply:
no I can't because binary data needs special handling which the generic function insertRow not supposed to know